lib/home/library/video_library.dart

```diff
@@ -1,30 +1,49 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
-import 'package:flutter/material.dart';
-import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:firebase_storage/firebase_storage.dart';
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:thc/firebase/firebase.dart';
 import 'package:thc/home/library/play_video.dart';
+import 'package:thc/utils/navigator.dart';
+import 'package:thc/utils/style_text.dart';
 import 'package:thc/utils/theme.dart';
 import 'package:intl/intl.dart';
 
 class VideoCard extends StatelessWidget {
-  const VideoCard(
-      {super.key,
-      required this.title,
-      required this.timestamp,
-      this.thumbnail,
-      required this.director,
-      this.category,
-      this.id,
-      this.path});
+  const VideoCard({
+    super.key,
+    required this.title,
+    required this.timestamp,
+    required this.director,
+    required this.category,
+    required this.id,
+    required this.path,
+    this.thumbnail,
+  });
+
   final String title;
   final Timestamp timestamp;
-  final String? thumbnail;
   final String director;
-  final String? path;
-  final category;
-  final id;
+  final String category;
+  final String id;
+  final String path;
+  final String? thumbnail;
+
+  Future<String?> playVideo() async {
+    if (!kIsWeb && defaultTargetPlatform == TargetPlatform.windows) {
+      return 'not supported on Windows';
+    }
+
+    String? url;
+    try {
+      final storageReference = FirebaseStorage.instance.ref().child(path);
+      url = await storageReference.getDownloadURL();
+    } catch (e) {
+      return '${e.runtimeType} has occurred: $e';
+    }
+    navigator.push(PlayVideo(videoURL: url, videoName: title));
+    return null;
+  }
```

Comment:
I thought moving the `onTap` out of the build method made it just a bit easier to read.

(also refactored it to show a `SnackBar` when there's an error instead of navigating to a text widget)

Reply:
Good cal. I like how you added the try catch for the url so if there is an error there we can catch it.